File: AssetManager/Helpers/DataGridHelpers/ColumnClasses.vb
Public Class DataGridColumn
    Public Property ColumnName As String
    Public Property ColumnCaption As String
    Public Property ColumnType As Type
    Public Property ColumnReadOnly As Boolean
    Public Property ColumnVisible As Boolean
    Public Property AttributeIndex As AttributeDataStruct()
    Public Property ColumnFormatType As ColumnFormatTypes

    Sub New(colName As String, caption As String, type As Type)
        ColumnName = colName
        ColumnCaption = caption
        ColumnType = type
        ColumnReadOnly = False
        ColumnVisible = True
        AttributeIndex = Nothing
        ColumnFormatType = ColumnFormatTypes.DefaultFormat
    End Sub

    Sub New(colName As String, caption As String, type As Type, displayMode As ColumnFormatTypes)
        ColumnName = colName
        ColumnCaption = caption
        ColumnType = type
        ColumnReadOnly = False
        ColumnVisible = True
        AttributeIndex = Nothing
        ColumnFormatType = displayMode
    End Sub

    Sub New(colName As String, caption As String, attribIndex() As AttributeDataStruct)
        ColumnName = colName
        ColumnCaption = caption
        ColumnType = GetType(String)
        ColumnReadOnly = False
        ColumnVisible = True
        Me.AttributeIndex = attribIndex
        ColumnFormatType = ColumnFormatTypes.AttributeCombo
    End Sub

    Sub New(colName As String, caption As String, attribIndex() As AttributeDataStruct, displayMode As ColumnFormatTypes)
        ColumnName = colName
        ColumnCaption = caption
        ColumnType = GetType(String)
        ColumnReadOnly = False
        ColumnVisible = True
        Me.AttributeIndex = attribIndex
        ColumnFormatType = displayMode
    End Sub

    Sub New(colName As String, caption As String, type As Type, isReadOnly As Boolean, visible As Boolean)
        ColumnName = colName
        ColumnCaption = caption
        ColumnType = type
        ColumnReadOnly = isReadOnly
        ColumnVisible = visible
        AttributeIndex = Nothing
        ColumnFormatType = ColumnFormatTypes.DefaultFormat
    End Sub

End Class

Public Enum ColumnFormatTypes
    DefaultFormat
    AttributeCombo
    AttributeDisplayMemberOnly
    NotePreview
    Image
    FileSize

End Enum

Public Structure StatusColumnColorStruct
    Public StatusID As String
    Public StatusColor As Color

    Sub New(id As String, color As Color)
        StatusID = id
        StatusColor = color
    End Sub

End Structure

File: AssetManager/Helpers/DataGridHelpers/ColumnClasses.cs
using System;
using System.Drawing;

namespace AssetManager
{
    public class DataGridColumn
    {
        public string ColumnName { get; set; }
        public string ColumnCaption { get; set; }
        public Type ColumnType { get; set; }
        public bool ColumnReadOnly { get; set; }
        public bool ColumnVisible { get; set; }
        public AttributeDataStruct[] AttributeIndex { get; set; }
        public ColumnFormatTypes ColumnFormatType { get; set; }

        public DataGridColumn(string colName, string caption, Type type)
        {
            ColumnName = colName;
            ColumnCaption = caption;
            ColumnType = type;
            ColumnReadOnly = false;
            ColumnVisible = true;
            AttributeIndex = null;
            ColumnFormatType = ColumnFormatTypes.DefaultFormat;
        }

        public DataGridColumn(string colName, string caption, Type type, ColumnFormatTypes displayMode)
        {
            ColumnName = colName;
            ColumnCaption = caption;
            ColumnType = type;
            ColumnReadOnly = false;
            ColumnVisible = true;
            AttributeIndex = null;
            ColumnFormatType = displayMode;
        }

        public DataGridColumn(string colName, string caption, AttributeDataStruct[] attribIndex)
        {
            ColumnName = colName;
            ColumnCaption = caption;
            ColumnType = typeof(string);
            ColumnReadOnly = false;
            ColumnVisible = true;
            this.AttributeIndex = attribIndex;
            ColumnFormatType = ColumnFormatTypes.AttributeCombo;
        }

        public DataGridColumn(string colName, string caption, AttributeDataStruct[] attribIndex, ColumnFormatTypes displayMode)
        {
            ColumnName = colName;
            ColumnCaption = caption;
            ColumnType = typeof(string);
            ColumnReadOnly = false;
            ColumnVisible = true;
            this.AttributeIndex = attribIndex;
            ColumnFormatType = displayMode;
        }

        public DataGridColumn(string colName, string caption, Type type, bool isReadOnly, bool visible)
        {
            ColumnName = colName;
            ColumnCaption = caption;
            ColumnType = type;
            ColumnReadOnly = isReadOnly;
            ColumnVisible = visible;
            AttributeIndex = null;
            ColumnFormatType = ColumnFormatTypes.DefaultFormat;
        }
    }
}

namespace AssetManager
{
    public enum ColumnFormatTypes
    {
        DefaultFormat,
        AttributeCombo,
        AttributeDisplayMemberOnly,
        NotePreview,
        Image,
        FileSize
    }
}

namespace AssetManager
{
    public struct StatusColumnColorStruct
    {
        public string StatusID;

        public Color StatusColor;

        public StatusColumnColorStruct(string id, Color color)
        {
            StatusID = id;
            StatusColor = color;
        }
    }
}